source/grid/grid_in.cc

```diff
@@ -1451,14 +1452,72 @@ GridIn<dim, spacedim>::read_msh(std::istream &in)
           in >> line;
         }
 
-      // if the next block is of kind $Entities, ignore it
+      // if the next block is of kind $Entities, parse it
       if (line == "$Entities")
         {
-          do
+          unsigned long n_points, n_curves, n_surfaces, n_volumes, n_physicals;
+          int           tag, physical_tag;
+          double        box_min_x, box_min_y, box_min_z, box_max_x, box_max_y,
+            box_max_z;
+
+          // we only care for physical_tag to fill tag_maps
+          in >> n_points >> n_curves >> n_surfaces >> n_volumes;
+          for (unsigned int i = 0; i < n_points; ++i)
             {
-              in >> line;
+              // parse point ids
+              in >> tag >> box_min_x >> box_min_y >> box_min_z >> box_max_x >>
+                box_max_y >> box_max_z >> n_physicals;
+              AssertThrow(n_physicals < 2,
+                          ExcMessage("More than one tag is not supported!"));
+              for (unsigned int j = 0; j < n_physicals; ++j)
+                in >> physical_tag;
+              tag_maps[0][tag] = (n_physicals == 0) ? 0 : physical_tag;
```

Comment:
So you're taking the last tag there is, or zero if there is none. Is this the correct choice? I have no idea what these tags do.

In any case, maybe you could add a comment here to state what I have in my first sentence and explain why.

Reply:
In all the files I have converted, there was never more than one "physical tag" and if one was present it matched the boundary id we should use later. If there is none, we use 0 as default
in conformance with the default boundary id. I don't know what to do if there is more than one tag. That's the reason for the `Assert`.

I added at least one comment saying where these tags will be used.

Reply:
Good enough -- we can always deal with the case of what to do if there are more than one tag when someone shows us an input file :-)

Reply:
... and tells us what to do with it.

Reply:
That too, yes :-)